### PR TITLE
Starting server plugin best practice doc

### PR DESCRIPTION
**Pull request by jasonblais**
This is to cover docs for https://github.com/mattermost/mattermost-server/pull/10476

### CODE REVIEW

**Comment by jasonblais** [comment]
@hanzei please help with initial review.

File: site/content/extend/plugins/server/best-practices.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+---
+title: Best Practices
+date: 2018-07-10T00:00:00-05:00
+subsection: Server Plugins
+weight: 0
+---
+
+## How should server plugins handle static files?
```
Comment (hanzei) [inline site/content/extend/plugins/server/best-practices.md]: The title is a bit ambiguous. There two cases if static files:
1. The plugins wants to load a static file and use it from the code
2. The file should be publicly available via http.

The text you proposed deals with 2. But the title doesn't clearly state this.

File: site/content/extend/plugins/server/best-practices.md [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+Add all static files under a file directory named `public` within the plugin directory, and include it in the plugin's Makefile.
```
Comment (hanzei) [inline site/content/extend/plugins/server/best-practices.md]: The sentence is missing a verb :wink:

Comment (hanzei) [inline site/content/extend/plugins/server/best-practices.md]: You want to include the files in the plugin bundle using the Makefile.

File: site/content/extend/plugins/server/best-practices.md [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+date: 2018-07-10T00:00:00-05:00
+subsection: Server Plugins
+weight: 0
+---
+
+## How should server plugins handle static files?
+
+Add all static files under a file directory named `public` within the plugin directory, and include it in the plugin's Makefile. 
+
+Then use `ServePluginPublicRequest` to serve the public plugin files at http(s)://$SITE_URL/plugins/$PLUGIN_ID/public/{your-static-file}`.
```
Comment (hanzei) [inline site/content/extend/plugins/server/best-practices.md]: Plugins don't have to do anything. The files are served automatically by the sever.

File: site/content/extend/plugins/server/best-practices.md [hunk 4]
```diff
@@ -0,0 +1,12 @@
+---
+title: Best Practices
+date: 2018-07-10T00:00:00-05:00
```
Comment (hanzei) [inline site/content/extend/plugins/server/best-practices.md]: Nit: I would remove the date, as it doesn't provide any information.

**Comment by jasonblais** [comment]
@hanzei addressed some of the feedback, would you be open to helping with the first two?

File: site/content/extend/plugins/server/best-practices.md (hunk 1 above)
Comment (hanzei) [reply site/content/extend/plugins/server/best-practices.md]: What do you think about: `How should plugins serve publicly available static files`?

Comment (jasonblais) [reply site/content/extend/plugins/server/best-practices.md]: Sounds good to me 👍

**Comment by jasonblais** [comment]
Merging in preparation for June 16th release. Changes also merged to demo and sample plugin.

**Pull request merged**